examples/nlp/pretraining_BERT.py

```diff
@@ -0,0 +1,481 @@
+(including converting the tokens to their corresponding IDs in the pretrained vocabulary)
+and put it in a format the model expects, as well as generate the other inputs that model
+requires.
+
+First we will make a list of all the raw documents from the `WikiText` corpus:
+"""
+
+all_texts = [
+    doc for doc in dataset["text"] if len(doc) > 0 and not doc.startswith(" =")
+]
+
+"""
+Next we will make a `batch_iterator` function that will aid us to train our tokenizer.
+"""
+
+
+def batch_iterator():
+    for i in range(0, len(all_texts), TOKENIZER_BATCH_SIZE):
+        yield all_texts[i : i + TOKENIZER_BATCH_SIZE]
+
+
+"""
+In this notebook, we will train a tokenizer with the exact same algorithms and
+parameters as an existing one. For instance, we will train a new version of the
+`BERT-CASED` tokenzier on `Wikitext-2` using the same tokenization algorithm.
+
+First we need to load the tokenizer we want to use as a model:
+"""
+
+from transformers import AutoTokenizer
+
+tokenizer = AutoTokenizer.from_pretrained(MODEL_CHECKPOINT)
+
+"""
+Now we will train our tokenizer using the entire `train` split of the `Wikitext-2`
+dataset.
+"""
+
+tokenizer = tokenizer.train_new_from_iterator(
+    batch_iterator(), vocab_size=TOKENIZER_VOCABULARY
+)
+
+"""
+So now we our done training our new tokenizer! Next we will move on to the data
+pre-processing steps.
+"""
+
+"""
+## Data Pre-processing
+"""
+
+"""
+For the sake of demonstrating the workflow, in this notebook we will only take
+small subset of the entire WikiText dataset.
+"""
+
+dataset = dataset.select([i for i in range(1000)])
+
+"""
+Before we can feed those texts to our model, we need to pre-process them and get them
+ready for the task. As mentioned earlier, the BERT pre-training task includes two tasks
+in total, the `NSP` task and the `MLM` task. 🤗 Transformers have an easy to implement
+`collator` called the `DataCollatorForLanguageModeling`. However, we need to get the
+data ready for `NSP` manually.
+
+
+Next we will write a simple function called the `prepare_train_features` that helps us in
+the pre-processing and is compatible with 🤗 Datasets. To summarize, our pre-processing
+function should:
+
+- Get the dataset ready for the NSP task by creating pairs of sentences (A,B), where B
+either actually follows A, or B is randomly sampled from somewhere else in the corpus.
+It should also generate a corresponding label for each pair, which is 1 if B actually
+follows A and 0 if not.
+- Tokenize the text dataset into it's corresponding token ids that will be used for
+embedding look-up in BERT
+- Create additional inputs for the model like `token_type_ids`, `attention_mask`, etc.
+"""
+
+# We define the maximum number of tokens after tokenization that each training sample
+# will have
+max_num_tokens = BLOCK_SIZE - tokenizer.num_special_tokens_to_add(pair=True)
+
+
+def prepare_train_features(examples):
+
+    """
```

Comment:
Please use standard Keras formatting for docstrings

Reply:
Completed this!